services/mirror/mirror_pull.go

```diff
@@ -235,6 +235,20 @@ func pruneBrokenReferences(ctx context.Context,
 	return pruneErr
 }
 
+// checkRecoverableSyncError takes an error message from a git fetch command and returns false if it should be a fatal/blocking error
+func checkRecoverableSyncError(stderrMessage string) bool {
+	switch {
+	case strings.Contains(stderrMessage, "unable to resolve reference") && strings.Contains(stderrMessage, "reference broken"):
+	case strings.Contains(stderrMessage, "remote error") && strings.Contains(stderrMessage, "not our ref"):
+	case strings.Contains(stderrMessage, "cannot lock ref") && strings.Contains(stderrMessage, "but expected"):
+	case strings.Contains(stderrMessage, "Unable to create") && strings.Contains(stderrMessage, ".lock"):
+	default:
+		return true
```

Comment:
It doesn't seem right. Golang `switch` doesn't "fallthrough" by default.

It's better to create a test case for this function, and add comments for each case: why it would happen and why it could be ignored.

Reply:
thanks for the feedback, I documented the existing logic in a test, along with the new cases of it I've found helpful at my company.